### ISSUE TITLE
No dimension set for key window

### ISSUE THREAD

**Issue by xiaoyongchen** [Microsoft/react-native-code-push#954]
### Description

com.facebook.react.modules.core.ExceptionsManagerModule.showOrThrowError(ExceptionsManagerModule.java:99)
--
2 com.facebook.react.modules.core.ExceptionsManagerModule.reportFatalException(ExceptionsManagerModule.java:83)
3 java.lang.reflect.Method.invoke(Native Method)
4 java.lang.reflect.Method.invoke(Method.java:372)
5 com.facebook.react.cxxbridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
6 com.facebook.react.cxxbridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:162)
7 com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
8 android.os.Handler.handleCallback(Handler.java:739)
9 android.os.Handler.dispatchMessage(Handler.java:95)
10 com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
11 android.os.Looper.loop(Looper.java:135)
12 com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:199)
13 java.lang.Thread.run(Thread.java:818)

### Additional Information

* react-native-code-push version:"react-native-code-push": "^4.1.0-beta",
* react-native version:"react-native": "0.46.4",
* iOS/Android/Windows version:Android
* Does this reproduce on a debug build or release build?release
* Does this reproduce on a simulator, or only on a physical device?

Android 7.0,level 24
HuaWei/EMOTION
arm64-v8a

**Comment by xiaoyongchen** [comment]
The use of this code resulted in a burst

**Comment by xiaoyongchen** [comment]
AppState.addEventListener("change", (newState) => {
    newState === "active" && CodePush.sync({
        installMode: CodePush.InstallMode.ON_NEXT_RESUME,
        minimumBackgroundDuration: 1 * 1
    });
});
CodePush.sync({installMode: CodePush.InstallMode.ON_NEXT_RESUME, minimumBackgroundDuration: 1 * 1});

**Comment by xiaoyongchen** [comment]
only one crash

**Comment by max-mironov** [comment]
hi @xiaoyongchen and thanks for the question. Could you please be a little bit more descriptive about the symptoms of the problem and provide us the following information to make it easier for us to determine the root cause of the issue:
1) Explain step by step what you did, what you expected to happen, and what actually happens. 
2) Could you try to reproduce the issue with our [Sample app](https://github.com/Microsoft/react-native-code-push/tree/master/Examples/CodePushDemoApp) (with the code your have shared with us?) Or the issue is only reproducible on your app?
3) In the closed duplicated item #934 it looks like the issue was related to iOS platform. Can you confirm that the issue is relevant to both platforms?
4) Can you explain what do you mean by "only one crash"? Does it mean that have seen that issue only once or the issue has stable reproduction steps?

**Comment by xiaoyongchen** [comment]
hi@max-mironov and thanks for your answer.This is the production environment, the platform is Andriod。It's just a problem caused by the upgrade ，react-native@0.43.3 ->react-native@0.46.4.

**Comment by xiaoyongchen** [comment]
Execute the code。

**Comment by xiaoyongchen** [comment]
Execute the code，AppState.addEventListener("change", (newState) => {
newState === "active" && CodePush.sync({
installMode: CodePush.InstallMode.ON_NEXT_RESUME,
minimumBackgroundDuration: 1 * 1
});
});
CodePush.sync({installMode: CodePush.InstallMode.ON_NEXT_RESUME, minimumBackgroundDuration: 1 * 1});

**Comment by max-mironov** [comment]
@xiaoyongchen understood and thanks. I've tried it on my side with RN 0.46.4 and RNCP 4.1.0-beta and and seems that it is working great. 
So seems that it could be and issue with upgrading to newer version of react-native.
Could you please try to do the following:
1) Completely remove react-native-code-push plugin
2) Launch the app without plugin to verify that the issue is not related to it
3) Add the plugin again from the scratch

Please let us know if it helps.

**Comment by max-mironov** [comment]
@xiaoyongchen - hi, do you have a chance to look at this?

**Comment by xiaoyongchen** [comment]
"Launch the app without plugin to verify that the issue is not related to it",is working.

**Comment by xiaoyongchen** [comment]
but ,The original project has not changed。

**Comment by xiaoyongchen** [comment]
@max-mironov  hi，Nothing much has changed。
Crash diary：'No dimension set for key window','Module AppRegistry is not a registered callable module ( calling runApplication)'

**Comment by xiaoyongchen** [comment]
graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
--
108-11 11:13:27.169 14289 14289 I MultiDex: VM with version 2.1.0 has multidex support
208-11 11:13:27.169 14289 14289 I MultiDex: install
308-11 11:13:27.169 14289 14289 I MultiDex: VM has multidex support, MultiDex support library is disabled.
408-11 11:13:27.179 14179 14179 D com.umeng.message.proguard.l: appLaunchAt=1502421181373
508-11 11:13:27.219 14289 14319 V xiaomi : [Thread:3861] JOB: prepare the mi push account.
608-11 11:13:27.229 14179 14312 E UClient : org.json.JSONException: Value ok of type java.lang.String cannot be converted to JSONObject: [Caught Exception in decryptData()]
708-11 11:13:27.229 14179 14312 E UClient : at	org.json.JSON.typeMismatch(JSON.java:111)
808-11 11:13:27.229 14179 14312 E UClient : at	org.json.JSONObject.<init>(JSONObject.java:160)
908-11 11:13:27.229 14179 14312 E UClient : at	org.json.JSONObject.<init>(JSONObject.java:173)
1008-11 11:13:27.229 14179 14312 E UClient : at	com.umeng.socialize.net.utils.UClient.decryptData(UClient.java:477)
1108-11 11:13:27.229 14179 14312 E UClient : at	com.umeng.socialize.net.utils.UClient.parseResult(UClient.java:410)
1208-11 11:13:27.229 14179 14312 E UClient : at	com.umeng.socialize.net.utils.UClient.httpGetRequest(UClient.java:213)
1308-11 11:13:27.229 14179 14312 E UClient : at	com.umeng.socialize.net.utils.UClient.execute(UClient.java:72)
1408-11 11:13:27.229 14179 14312 E UClient : at	com.umeng.socialize.net.base.SocializeClient.execute(SocializeClient.java:33)
1508-11 11:13:27.229 14179 14312 E UClient : at	com.umeng.socialize.net.stats.StatsAPIs.dauStats(StatsAPIs.java:124)
1608-11 11:13:27.229 14179 14312 E UClient : at	com.umeng.socialize.net.analytics.SocialAnalytics$8.run(SocialAnalytics.java:238)
1708-11 11:13:27.229 14179 14312 E UClient : at	java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
1808-11 11:13:27.229 14179 14312 E UClient : at	java.util.concurrent.FutureTask.run(FutureTask.java:237)
1908-11 11:13:27.229 14179 14312 E UClient : at	java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
2008-11 11:13:27.229 14179 14312 E UClient : at	java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
2108-11 11:13:27.229 14179 14312 E UClient : at	java.lang.Thread.run(Thread.java:818)
2208-11 11:13:27.229 14289 14289 V xiaomi : [Thread:1] XMPushService created pid = 14289
2308-11 11:13:27.229 14289 14289 V xiaomi : [Thread:1] network changed, [type: WIFI[], state: CONNECTED/CONNECTED, reason: (unspecified), extra: "zmlearn", roaming: false, failover: false, isAvailable: true, isConnectedToProvisioningNetwork: false]
2408-11 11:13:27.239 14289 14319 V xiaomi : [Thread:3861] JOB: do reconnect..
2508-11 11:13:27.239 14289 14319 V xiaomi : [Thread:3861] update the connection status. disconnected -> connecting : ERROR_OK
2608-11 11:13:27.249 14289 14319 V xiaomi : [Thread:3861] begin to connect to 203.100.92.12:5222
2708-11 11:13:27.259 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:origin statusbar style
2808-11 11:13:27.259 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:windowDrawsFlag set
2908-11 11:13:27.259 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:IconColor=1
3008-11 11:13:27.259 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:StatusBarColor final set ff757575
3108-11 11:13:27.289 14179 14198 D [ui]AppForegroundWatcher: 1/app on foreground
3208-11 11:13:27.289 14289 14319 V xiaomi : [Thread:3861] tcp connected
3308-11 11:13:27.299 14289 14319 V xiaomi : [Thread:3861] load config failure: /data/data/com.zmlearn.course.am/files/XMCloudCfg: open failed: ENOENT (No such file or directory)
3408-11 11:13:27.299 14289 14319 V xiaomi : [Thread:3861] [slim] open conn: andver=22 sdk=26 hash=N5hOnp4u_W2Rq0VI tz=8:0 Model=vivo X6S A os=eng.compiler.20170705.161444
3508-11 11:13:27.299 14289 14319 V xiaomi : [Thread:3861] connected to 203.100.92.12:5222 in 45
3608-11 11:13:27.299 14179 14227 I TDLog : onResume being called! pageName: login.WelcomeActivity, FromAPI: false
3708-11 11:13:27.299 14179 14234 I TDLog : [Session] - Same session as before!
3808-11 11:13:27.319 14179 14332 D SensorManager: open file /sys/networktype/networktype error
3908-11 11:13:27.319 14179 14332 D SensorManager: isQQShake: package is = com.zmlearn.course.am
4008-11 11:13:27.319 14179 14332 D Sensors : enableSensor handle 1 20000us 0us in
4108-11 11:13:27.319 14179 14332 D Sensors : enableSensor handle 1 ont
4208-11 11:13:27.319 14179 14179 I ViewRootImpl: CPU Rendering VSync enable = true
4308-11 11:13:27.319 14179 14339 D OpenGLRenderer: Use EGL_SWAP_BEHAVIOR_PRESERVED: true
4408-11 11:13:27.329 14179 14198 I [ui]main_conn: 1/onServiceConnected: binder#android.os.BinderProxy@3b831f8
4508-11 11:13:27.329 14200 14220 I [push]remote: 3861/IPC duplex channel established
4608-11 11:13:27.329 14179 14198 I [ui]aux_conn: 1/onServiceConnected: binder#android.os.BinderProxy@127706d1
4708-11 11:13:27.329 14179 14198 I [ui]ui : 1/Push awake UI by Broadcast success, UI process run!
4808-11 11:13:27.329 14179 14340 D com.umeng.message.UmengMessageCallbackHandlerService: processName=com.zmlearn.course.am
4908-11 11:13:27.329 14179 14340 D com.umeng.message.UmengMessageCallbackHandlerService: enable(): register-->:AvJOt956fyWr9XA5FtMd2yI_EYNBq9FrjK79AThTaird,status:true
5008-11 11:13:27.329 14179 14340 E PushProcess umeng push device_token:: com.zmlearn.course.am.application.ZMLearnCourseAmApplication$2::onSuccess@374>>>AvJOt956fyWr9XA5FtMd2yI_EYNBq9FrjK79AThTaird
5108-11 11:13:27.339 14179 14198 I [ui]ui : 3853/set status from UNLOGIN to UNLOGIN
5208-11 11:13:27.339 14179 14198 I [ui]InvocationMgr: 3853/on notify: Transaction: [id: 2, method: AuthServiceObserver/observeOnlineStatus]
5308-11 11:13:27.339 14179 14198 I [ui]InvocationMgr: 3853/on notify: Transaction: [id: 3, method: AuthServiceObserver/observeOtherClients]
5408-11 11:13:27.349 14289 14328 V xiaomi : [Thread:3869] setChallenge hash = 14b12023
5508-11 11:13:27.349 14289 14328 V xiaomi : [Thread:3869] update the connection status. connecting -> connected : ERROR_OK
5608-11 11:13:27.349 14179 14339 I Adreno : QUALCOMM build : 814500f, If3abccc389
5708-11 11:13:27.349 14179 14339 I Adreno : Build Date : 03/15/17
5808-11 11:13:27.349 14179 14339 I Adreno : OpenGL ES Shader Compiler Version: XE031.06.00.03
5908-11 11:13:27.349 14179 14339 I Adreno : Local Branch :
6008-11 11:13:27.349 14179 14339 I Adreno : Remote Branch : refs/tags/AU_LINUX_ANDROID_LA.BR.1.3.2.C2.05.01.01.199.103
6108-11 11:13:27.349 14179 14339 I Adreno : Remote Branch : NONE
6208-11 11:13:27.349 14179 14339 I Adreno : Reconstruct Branch : NOTHING
6308-11 11:13:27.349 14179 14339 I OpenGLRenderer: Initialized EGL, version 1.4
6408-11 11:13:27.349 14289 14319 V xiaomi : [Thread:3861] JOB: bind the client. 5
6508-11 11:13:27.349 14289 14319 V xiaomi : [Thread:3861] update the client 5 status. unbind->binding unknown ERROR_OK null null
6608-11 11:13:27.349 14289 14328 V xiaomi : [Thread:3869] [Slim] CONN: host = 4648
6708-11 11:13:27.349 14289 14319 V xiaomi : [Thread:3861] [Slim]: bind id=7bnMD-1
6808-11 11:13:27.349 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
6908-11 11:13:27.359 14179 14339 D OpenGLRenderer: Enabling debug mode 0
7008-11 11:13:27.359 14179 14345 I XiaoMiPushMsgReceiver: onCommandResult:command={register}, resultCode={0}, reason={null}, category={null}, commandArguments={[6Y69FJ538IGna9tEdSblE6XnRY4cY6miVk4V1Ri9UsA=]}
7108-11 11:13:27.359 14179 14345 I XiaoMiPushMsgReceiver: onReceiveRegisterResult:command={register}, resultCode={0}, reason={null}, category={null}, commandArguments={[6Y69FJ538IGna9tEdSblE6XnRY4cY6miVk4V1Ri9UsA=]}
7208-11 11:13:27.359 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
7308-11 11:13:27.399 14179 14198 I [ui]NOS_LBS: 1/load cached nos upload server addresses from SP, ip count=1, default ip count=1
7408-11 11:13:27.409 14289 14319 V xiaomi : [Thread:3861] SMACK: channel bind succeeded, chid=5
7508-11 11:13:27.409 14289 14319 V xiaomi : [Thread:3861] update the client 5 status. binding->binded OPEN ERROR_OK null null
7608-11 11:13:27.729 14179 14179 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@1a00ede5 time:143461659
7708-11 11:13:28.339 14179 14229 I TDLog : Data submitting Succeed!
7808-11 11:13:31.349 14179 14179 I Timeline: Timeline: Activity_launch_request id:com.zmlearn.course.am time:143465279
7908-11 11:13:31.379 14179 14227 I TDLog : onPause being called! pageName: login.WelcomeActivity
8008-11 11:13:31.389 14179 14179 V ActivityThread: Handle configuration changed: {1.0 ?mcc?mnc zh_CN ?layoutDir sw360dp w360dp h616dp 480dpi nrml long port finger -keyb/v/h -nav/h s.37 themeId=0, affectGlobal:true}
8108-11 11:13:31.389 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
8208-11 11:13:31.389 14179 14179 V ActivityThread: Launching activity com.zmlearn.course.am.framework.FrameActivity with config {1.0 ?mcc?mnc zh_CN ?layoutDir sw360dp w360dp h616dp 480dpi nrml long port finger -keyb/v/h -nav/h s.37 themeId=0, affectGlobal:true}
8308-11 11:13:31.409 14179 14179 D com.umeng.message.proguard.l: appLaunchAt=1502421181373
8408-11 11:13:31.439 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:origin statusbar style
8508-11 11:13:31.439 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:windowDrawsFlag set
8608-11 11:13:31.439 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:IconColor=1
8708-11 11:13:31.439 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:StatusBarColor final set ff757575
8808-11 11:13:31.439 14179 14179 E FrameActivity: com.zmlearn.course.am.framework.FrameActivity::onCreate@173>>>umeng push device_token:AvJOt956fyWr9XA5FtMd2yI_EYNBq9FrjK79AThTaird
8908-11 11:13:31.459 14179 14179 D ReactNative: [CodePush] Loading JS bundle from "/data/data/com.zmlearn.course.am/files/CodePush/0922746b93cd127042ea27c08146ee7d911b9b7fe873e413858ad2fab958e6b8/CodePush/index.android.bundle"
9008-11 11:13:31.459 14179 14179 D ReactNative: [CodePush] Loading JS bundle from "/data/data/com.zmlearn.course.am/files/CodePush/0922746b93cd127042ea27c08146ee7d911b9b7fe873e413858ad2fab958e6b8/CodePush/index.android.bundle"
9108-11 11:13:31.589 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:origin statusbar style
9208-11 11:13:31.589 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:IconColor=1
9308-11 11:13:31.589 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:StatusBarColor final set ff757575
9408-11 11:13:31.599 14179 14179 I ViewRootImpl: CPU Rendering VSync enable = true
9508-11 11:13:31.679 14179 14227 I TDLog : onResume being called! pageName: framework.FrameActivity, FromAPI: false
9608-11 11:13:31.699 14179 14234 I TDLog : [Session] - Same session as before!
9708-11 11:13:31.709 14179 14179 I ViewRootImpl: CPU Rendering VSync enable = true
9808-11 11:13:31.739 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
9908-11 11:13:31.759 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
10008-11 11:13:31.769 14179 14179 E TEST : onPageSelected
10108-11 11:13:31.889 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
10208-11 11:13:31.919 14179 14179 E TEST : onPageSelected
10308-11 11:13:31.949 14179 14179 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@1fede475 time:143465871
10408-11 11:13:33.229 14179 14179 I Timeline: Timeline: Activity_launch_request id:com.zmlearn.course.am time:143467155
10508-11 11:13:33.249 14179 14227 I TDLog : onEvent being called! eventId: 1_lunbo1gengduo, eventLabel: , eventMap: mapSize: 1
10608-11 11:13:33.259 14179 14227 I TDLog : onPause being called! pageName: framework.FrameActivity
10708-11 11:13:33.279 14179 14179 V ActivityThread: Handle configuration changed: {1.0 ?mcc?mnc zh_CN ?layoutDir sw360dp w360dp h616dp 480dpi nrml long port finger -keyb/v/h -nav/h s.37 themeId=0, affectGlobal:true}
10808-11 11:13:33.279 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
10908-11 11:13:33.279 14179 14179 V ActivityThread: Launching activity com.zmlearn.course.am.react.view.ReactViewContainActivity with config {1.0 ?mcc?mnc zh_CN ?layoutDir sw360dp w360dp h616dp 480dpi nrml long port finger -keyb/v/h -nav/h s.37 themeId=0, affectGlobal:true}
11008-11 11:13:33.299 14179 14179 I art : Thread[1,tid=14179,WaitingForJniOnLoad,Thread*=0xab677160,peer=0x74c18e40,"main"] recursive attempt to load library "/data/app/com.zmlearn.course.am-1/lib/arm/libfb.so"
11108-11 11:13:33.309 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:origin statusbar style
11208-11 11:13:33.309 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:windowDrawsFlag set
11308-11 11:13:33.309 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:IconColor=1
11408-11 11:13:33.309 14179 14179 D PhoneWindow: DEBUG_SYSTEMUI:StatusBarColor final set ff757575
11508-11 11:13:33.319 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTGroupViewManager
11608-11 11:13:33.319 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTGroupShadowNode
11708-11 11:13:33.319 14179 14227 I TDLog : onResume being called! pageName: react.view.ReactViewContainActivity, FromAPI: false
11808-11 11:13:33.319 14179 14179 I ViewRootImpl: CPU Rendering VSync enable = true
11908-11 11:13:33.319 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTShapeViewManager
12008-11 11:13:33.329 14179 14234 I TDLog : [Session] - Same session as before!
12108-11 11:13:33.329 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTShapeShadowNode
12208-11 11:13:33.329 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTTextViewManager
12308-11 11:13:33.329 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTTextShadowNode
12408-11 11:13:33.329 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.picker.ReactDialogPickerManager
12508-11 11:13:33.329 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.uimanager.LayoutShadowNode
12608-11 11:13:33.339 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.drawer.ReactDrawerLayoutManager
12708-11 11:13:33.339 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.picker.ReactDropdownPickerManager
12808-11 11:13:33.339 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.scroll.ReactHorizontalScrollViewManager
12908-11 11:13:33.349 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.progressbar.ReactProgressBarViewManager
13008-11 11:13:33.349 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.progressbar.ProgressBarShadowNode
13108-11 11:13:33.349 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.scroll.ReactScrollViewManager
13208-11 11:13:33.349 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.slider.ReactSliderManager
13308-11 11:13:33.349 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.slider.ReactSliderManager$ReactSliderShadowNode
13408-11 11:13:33.349 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.switchview.ReactSwitchManager
13508-11 11:13:33.359 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.switchview.ReactSwitchManager$ReactSwitchShadowNode
13608-11 11:13:33.359 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.toolbar.ReactToolbarManager
13708-11 11:13:33.359 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.webview.ReactWebViewManager
13808-11 11:13:33.359 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.swiperefresh.SwipeRefreshLayoutManager
13908-11 11:13:33.359 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTSurfaceViewManager
14008-11 11:13:33.359 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.art.ARTSurfaceViewShadowNode
14108-11 11:13:33.359 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.frescosupport.FrescoBasedReactTextInlineImageViewManager
14208-11 11:13:33.359 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.frescosupport.FrescoBasedReactTextInlineImageShadowNode
14308-11 11:13:33.359 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
14408-11 11:13:33.369 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.image.ReactImageManager
14508-11 11:13:33.369 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.modal.ReactModalHostManager
14608-11 11:13:33.369 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.modal.ModalHostShadowNode
14708-11 11:13:33.369 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactRawTextManager
14808-11 11:13:33.369 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactTextShadowNode
14908-11 11:13:33.369 14179 14179 V ResourcesManager: Skipping new config: curSeq=37, newSeq=37
15008-11 11:13:33.379 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.textinput.ReactTextInputManager
15108-11 11:13:33.379 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.textinput.ReactTextInputShadowNode
15208-11 11:13:33.379 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactTextViewManager
15308-11 11:13:33.379 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.view.ReactViewManager
15408-11 11:13:33.379 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.viewpager.ReactViewPagerManager
15508-11 11:13:33.379 14179 14384 W unknown:ViewManagerPropertyUpdater: Could not find generated setter for class com.facebook.react.views.text.ReactVirtualTextViewManager
15608-11 11:13:33.399 14179 14339 D OpenGLRenderer: endAllStagingAnimators on 0xabc62818 (RippleDrawable) with handle 0xabc4d830
15708-11 11:13:33.409 14179 14179 I Timeline: Timeline: Activity_idle id: android.os.BinderProxy@264c6418 time:143467332
15808-11 11:13:33.629 14179 14386 E ReactNativeJS: No dimension set for key window
15908-11 11:13:33.639 14179 14386 E ReactNativeJS: Module AppRegistry is not a registered callable module (calling runApplication)

**Comment by max-mironov** [comment]
@xiaoyongchen - hi again. Is it possible for you to share with us the minimum app (link on github or probably a zip archive) so we can debug locally to get out what the issue is? 
As far as I can see at that moment there is something wrong with how you integrate code push but without the sample app it is really not easy to say what you are doing wrong.

**Comment by xiaoyongchen** [comment]
@max-mironov  ，What's your e-mail address?
Send you an e-mail

**Comment by xiaoyongchen** [comment]
@max-mironov  Please check the mail

**Comment by max-mironov** [comment]
@xiaoyongchen thanks, replied you via email.

**Comment by xiaoyongchen** [comment]
@max-mironov  There will still be a burst in the release environment

**Comment by xiaoyongchen** [comment]
@max-mironov  hi，'No dimension set for key window','Module AppRegistry is not a registered callable module ( calling runApplication)'
What causes this problem?

**Comment by xiaoyongchen** [comment]
Kill the process, try several times. There will be a run down problem

**Comment by xiaoyongchen** [comment]
Needs to be done in the production environment

**Comment by xiaoyongchen** [comment]
You need to kill in the production environment, kill the service, restart the project, and click the button several times

**Comment by xiaoyongchen** [comment]
@max-mironov  Did you find the reason?

**Comment by max-mironov** [comment]
Hi @xiaoyongchen.
Could you please try to compose your messages in one post so it would be easier to collaborate with this in future.

As for your issue, once again - I don't see any problems and how it is related with code-push there.
More than that I was able to run your project with release configuration (on Android real deive) without any issues (and also restart it several times successfully).

I've done the following:
1) Remove `node_modules`folder and `package-lock.json` file from the project
2) Remove everything from android/app/build folder
3) Run `npm install`
4) Execute `react-native run-android` and run your app successfully 
5) Also run your app with release build and was able to run the app successfully.

Anyway I don't see any issues here with code-push, if you see any problems with code-push could you try to be more specific about the symptoms and try to replicate them with fresh new app with the react-native and code-push version where you experience the issue. 
Otherwise I don't see any relevance here within our plugin.

Please let us know if any questions or considerations.

**Comment by xiaoyongchen** [comment]
Hi，@max-mironov.
Thank you for answering my question.
In the production environment. Do as you say, the project is a success. But after I kill the program, repeat the steps above. The question that I speak shows. The program broke down

**Comment by max-mironov** [comment]
@xiaoyongchen 
Hmm just have tried to kill the app on the device and reopen it again several times and didn't see any issues.
Anyway it looks like the issue described at the beginning of the thread is not reproduced anymore so I think this item could be closed as it is not a good practice to discuss and fix several items in one single thread. 

If you can see other problems that are indeed related to code-push I believe it make sense to create another issue and provide us with all the info that is described in our [new Issue Template](https://github.com/Microsoft/react-native-code-push/issues/new). Also it would be great if you could verify if you can see the issue with our [Sample App](https://github.com/Microsoft/react-native-code-push/tree/master/Examples/CodePushDemoApp).

**Comment by sospedra** [comment]
I ran into the same issue. Which causes android app to break only on release mode.

It only triggers the error after installation first run. But it's possible that, later, codepush doesn't work well (without triggering this error).

This could maybe be related with #968 
I find some relations:

* the release/debug mode
* only breaks the app after first installation

Currently we've uninstalled `code-push` from our project. We're gonna finish our product release today and start with the `code-push` issue on Monday; most probably.

I'll keep you posted. Tho, any tip would be much appreciated.

cc @carloscuesta

**Comment by max-mironov** [comment]
@sospedra that looks like something that require additional investigation. 
Could you try to be more descriptive about how to reproduce this issue e.g. by doing a) ... b) ... c) ... steps so we could replicate the issue on our side and see what's going wrong here?
Also if you could reproduce this case (by integrading code-push like you do in your real app) with `rninit init NewApp` it would be very helpful.

**Comment by xiaoyongchen** [comment]
@sospedra  Can you work on your project?
Is there a solution to this problem?

**Comment by carloscuesta** [comment]
Hey @xiaoyongchen, a clean reinstall of `react-native-code-push` fixed the issue.

**Comment by xiaoyongchen** [comment]
@carloscuesta  are you sure?

**Comment by xiaoyongchen** [comment]
Can solve the problem，When I reinstall the codepush

**Comment by Gnative** [comment]
Incase anyone else has this issue, check that your app is not loading an older bundle from Codepush that then causes this error.

In my case I rebuilt, upgraded my React Native App and re-incorporated Code Push. Which then on my Release version would pull in an old bundle and caused this error. An indicator might be that the error seems delayed after App initialisation. Which is when the Codepush bundle might be being applied.

My fix was to change the Deployment History versions to restrict the older versions to only be applied to a lower targetBinaryVersion of my RN App.
